doc/hld_template.md

```diff
@@ -0,0 +1,65 @@
+# HLD Name #
+
+## Table of Content 
+
+### 1. Revision  
```

Comment:
Remove the numbers - you have 2 sections 1, 2, and 3.

Reply:
Done.

Reply:
I see you removed all the numbers. Note that I'm not averse to numbers - I just want them to be correct/consistent.